joomla/joomla-cms
# [4.0] Multilingual status module improvements

Replaces https://github.com/joomla/joomla-cms/pull/24062 which had conflicts

This PR adds a Notice and displays the language concerned when the Content Language for that language has been deleted and remains in the site an orphan home page while the site language is published.
It also adds a Notice when a Content Language is in the trash.

It also reinstate all states of a content language to display the correct icons.
published = 1
published = 0
published = -2 (i.e. trash) and for this one displays the trash icon

It also modifies all the queries in the Helper to use quotename()

See more below

### Testing Instructions
Install a multilingual site with all languages available to install
French fr-FR
German de-DE
Persian fa-IR

Then use the Cpanel Multilingual Sample Data module to create a basic.

First test:
Display the Content languages Manager
Trash AND delete the fa-IR Persian content language.

You should get this when displaying the multilingual status module

<img width="1464" alt="screen shot 2019-03-02 at 18 25 09" src="https://user-images.githubusercontent.com/869724/53685264-a3e95c80-3d18-11e9-9722-8e0f1c2d1e62.png">

Then unpublish the French fr-FR Content Language.
And trash (do NOT delete) the German de-DE Content language.
I.e. Manager should look like this
<img width="671" alt="screen shot 2019-03-02 at 18 28 10" src="https://user-images.githubusercontent.com/869724/53685300-03476c80-3d19-11e9-84bc-7b05a02f92bc.png">

You should get this when displaying the multilingual status module
<img width="1486" alt="screen shot 2019-03-02 at 18 44 18" src="https://user-images.githubusercontent.com/869724/53685466-4e627f00-3d1b-11e9-9ac4-5bba6921afc8.png">

Unpublish also the fr-FR Home page to get a new notice.
Deleted Content Language error now displays the notice as last.
see https://github.com/joomla/joomla-cms/pull/24062#issuecomment-469004768

Added new Notice when both Content Language and Home page are unpublished.
We should now get in that case:

<img width="1286" alt="screen shot 2019-03-03 at 10 22 15" src="https://user-images.githubusercontent.com/869724/53693319-24eb3700-3d9f-11e9-9c3e-28000bfcd3e9.png">

 Added notice when both content language and home page missing. added trash icon when home page is in the trash.

<img width="1466" alt="screen shot 2019-03-04 at 12 40 18" src="https://user-images.githubusercontent.com/869724/53731340-be3b4b80-3e7a-11e9-996a-d914b053d918.png">

Added warning icon when home page does not exist.
<img width="1505" alt="screen shot 2019-03-04 at 12 56 37" src="https://user-images.githubusercontent.com/869724/53732061-ffccf600-3e7c-11e9-9c11-af880e58f3b7.png">

### Reminder
It is OK to have an Unpublished Content Language (as French is above) as it lets the site owner prepare a new Language for the multilingual site.


The screenshots were taken in March. They will now rather look like this:
<img width="1366" alt="Screen Shot 2019-11-30 at 08 44 07" src="https://user-images.githubusercontent.com/869724/69897531-a59a8c00-134d-11ea-89ea-a61dc4f5cf8e.png">

@Quy
@alikon 
Please test and specially look at the queries as I did my best but there could be improvements there.
